packages/block-library/src/navigation-link/edit.js

```diff
@@ -311,23 +303,35 @@ export default compose( [
 			getClientIdsOfDescendants,
 			hasSelectedInnerBlock,
 			getBlockParentsByBlockName,
+			getColorObjectByColorSlug,
 		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
 		const rootBlock = head(
 			getBlockParentsByBlockName( clientId, 'core/navigation' )
 		);
-		const navigationBlockAttributes = getBlockAttributes( rootBlock );
+		const navBlockAttrs = getBlockAttributes( rootBlock );
 		const hasDescendants = !! getClientIdsOfDescendants( [ clientId ] )
 			.length;
 		const showSubmenuIcon =
-			!! navigationBlockAttributes.showSubmenuIcon && hasDescendants;
+			!! navBlockAttrs.showSubmenuIcon && hasDescendants;
 		const isParentOfSelectedBlock = hasSelectedInnerBlock( clientId, true );
 
 		return {
 			isParentOfSelectedBlock,
 			hasDescendants,
 			showSubmenuIcon,
-			navigationBlockAttributes,
+			textColor: navBlockAttrs.textColor,
```

Comment:
Do the navigation items need to explicitly set the colors to have the same values as the parent? Could they have styles that inherit the color?

Reply:
I remember there have been problems with inheritance and not having enough power to override CSS rules of many themes. Bringing values inline this way resolved most of them. Maybe somebody else can explain more. 

The long run would be to possibly move to CSS variables but we cannot do it yet because of remaining IE support.

Reply:
Yes, it's necessary to apply the styles rightly to navigation links. Sometimes it isn't, depending on the type of color applied (from colors palette or custom one), or the theme, etc.
CSS vars simplifies this enormously, but again it isn't supported by IE.

Reply:
> Do the navigation items need to explicitly set the colors to have the same values as the parent? Could they have styles that inherit the color?

We apply the CSS class and inline style to each NavigationLink as the Navigation does. In some way, it follows up the behavior of the `useColors()` hook through its HOCs.
Also, it's aligned with the idea of being able to set up specific colors for each item.